src/Transformer/TransformerIterator.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace RebelCode\EddBookings\RestApi\Transformer;
+
+use Iterator;
+
+/**
+ * An iterator implementation that wraps and iterates over another iterator and applies a transformation on each
+ * element in the iteration before yielding.
+ *
+ * @since [*next-version*]
+ */
+class TransformerIterator implements Iterator
+{
+    /**
+     * The transformer to use.
+     *
+     * @since [*next-version*]
+     *
+     * @var TransformerInterface
+     */
+    protected $transformer;
+
+    /**
+     * The wrapped iterator.
+     *
+     * @since [*next-version*]
+     *
+     * @var Iterator
+     */
+    protected $iterator;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param TransformerInterface $transformer The transformer to use.
+     * @param Iterator             $iterator    The iterator to wrap.
```

Comment:
Does this *have* to be an iterator, or will any `Traversable` do? There's a method for normalization of this stuff.